sys/vm/vm_object.c

```diff
@@ -301,6 +301,14 @@ vm_object_init(void)
 	kernel_object->flags |= OBJ_COLORED;
 	kernel_object->pg_color = (u_short)atop(VM_MIN_KERNEL_ADDRESS);
 #endif
+#if __has_feature(capabilities)
+	/*
+	 * Let's be honest: the kernel stores capabilities.  This matters less
+	 * than one might think, tho', because the kernel page tables are often
+	 * built by locore or are otherwise specially handled.
+	 */
+	kernel_object->flags &= ~(OBJ_NOSTORETAGS | OBJ_NOLOADTAGS);
```

Comment:
So why are these set?  Is it because the kernel_object is OBJT_PHYS?  We probably should not be setting these flags on OBJT_PHYS in the first place if so.

Reply:
I didn't fiddle the creation of OBJT_PHYS objects because it seems like capability presence might be more per-object than per-type at that point?  But if you think that's the better fix, or if you'd prefer that tag storage were opt-out rather than opt-in, I can.

Reply:
I think not having it for `OBJT_PHYS`was probably oversight.  `OBJT_PHYS` is just `OBJT_DEFAULT` that can't be converted to swap.